keyboards/annepro2/keymaps/bla/keymap.c

```diff
@@ -0,0 +1,153 @@
+  * |-----------------------------------------------------------------------------------------+
+  * | Tab    |  q  | UP  |  e  |  r  |  t  |  y  |  u  |  i  |  o  | PS | HOME | END |   \    |
+  * |-----------------------------------------------------------------------------------------+
+  * | Esc     |LEFT |DOWN |RIGHT|  f  |  g  |  h  |  j  |  k  |  l  | PGUP|PGDN |    Enter    |
+  * |-----------------------------------------------------------------------------------------+
+  * | Shift      |  z  |  x  |  c  |  v  |  b  |  n  |  m  |  ,  |INSRT| DEL |    Shift       |
+  * |-----------------------------------------------------------------------------------------+
+  * | Ctrl  |  L1   |  Alt  |               space             |  Alt  |  FN1  |  FN2  | Ctrl  |
+  * \-----------------------------------------------------------------------------------------/
+  *
+  */
+ [_FN2_LY] = KEYMAP( /* Base */
+    KC_TRNS, KC_AP2_BT1, KC_AP2_BT2, KC_AP2_BT3, KC_AP2_BT4, KC_TRNS, KC_TRNS, KC_TRNS, KC_AP_LED_OFF, KC_AP_LED_ON, KC_AP_LED_NEXT_INTENSITY, KC_AP_LED_SPEED, KC_TRNS, KC_MEDIA_PLAY_PAUSE,
+    MO(_FN2_LY),    KC_TRNS, KC_UP,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PSCR, KC_HOME, KC_END, KC_TRNS,
+    KC_TRNS,        KC_LEFT, KC_DOWN, KC_RGHT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PGUP, KC_PGDN, KC_TRNS,
+    KC_TRNS,        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_INS,  KC_DEL,           KC_TRNS,
+    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, MO(_FN1_LY), MO(_FN2_LY), KC_TRNS
+ ),
+};
+const uint16_t keymaps_size = sizeof(keymaps);
+
+
+void matrix_init_user(void) {
+}
+
+void matrix_scan_user(void) {
+}
+
+// Code to run after initializing the keyboard
+void keyboard_post_init_user(void) {
+    // Here are two common functions that you can use. For more LED functions, refer to the file "qmk_ap2_led.h"
+
+    // annepro2-shine disables LEDs by default. Uncomment this function to enable them at startup.
+    // annepro2LedEnable();
+
+    // Additionally, it also chooses the first LED profile by default. Refer to the "profiles" array in main.c in
+    // annepro2-shine to see the order. Replace "i" with the index of your preferred profile. (i.e the RED profile is index 0)
+    // annepro2LedSetProfile(i);
+    debug_enable=true;
+}
+
+layer_state_t layer_state_set_user(layer_state_t layer) {
+  switch(get_highest_layer(layer)) {
+    case _FN1_LY:
+      // Set the leds to green
+      annepro2LedSetForegroundColor(0x00, 0xFF, 0x00);
+      break;
+    case _FN2_LY:
+      // Set the leds to blue
+      annepro2LedSetForegroundColor(0x00, 0x00, 0xFF);
+      break;
+    default:
+      // Reset back to the current profile
+      annepro2LedResetForegroundColor();
+      break;
+  }
+  return layer;
+}
+
+// The function to handle the caps lock logic
+bool led_update_user(led_t leds) {
```

Comment:
This causes the foreground change of a layer to NOT include the capslock - it's probably executed not only when the capslock is pressed/depressed.

Reply:
Fixed that.